### PR TITLE
[CI-5758] [CI-6410] [DOC-2351] - Connector settings, Connector authentication, Connectors in CI steps

### PR DESCRIPTION
**Pull request by aimurphy**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [x] Issue
- [x] Feature
- [x] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* CI-5758 : No specific change for this ticket, but applied general revisions to some GCP connector-related topics. I used the issue in this ticket for an example in the `GCP connector errors` part of `connect-to-google-cloud-platform-gcp.md`
* CI-6410 : Updated 9 step settings topics to include a note about which authentication methods are supported, not support, or untested (which I phrased as `may experience errors during pipeline runs`).
   * I only made the minimum edits to these topics since the PR already includes so many changes. 
   * Modified topics are the 'step setting' topics per the first column here: https://harness.atlassian.net/wiki/spaces/PD/pages/21254275563/Gaps+in+CI+Steps+with+Connectors 
   * EXCEPT:
      *  `run` step (which already got a great deal of changes from another PR and can use a variety of connectors; however, CI-5758 is included in this PR)
      * `build and push to acr` step (which is missing from the docs - I created DOC-2597 for this and I will address it if there is time before the release).
* DOC-2351 : upload-artifacts-to-s-3-step-settings.md : Add `Stage variable required for ARNs`
* Other topics also edited because they were referenced by topics I was already editing and they needed corresponding revisions to similar content

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/continuous-integration/ci-technical-reference/build-and-push-to-docker-hub-step-settings.md [hunk 1]
```diff
@@ -20,6 +20,12 @@ The unique name for this step.
 
 The Harness Docker Registry Connector to use for uploading the image. See [Docker Connector Settings Reference](../../platform/7_Connectors/ref-cloud-providers/docker-registry-connector-settings-reference.md).
 
+:::note
+
+This step supports Docker connectors that use either anonymous or username and password authentication.
+
+:::
+
```
Comment (aimurphy) [inline docs/continuous-integration/ci-technical-reference/build-and-push-to-docker-hub-step-settings.md:28]: CI-6410

File: docs/continuous-integration/ci-technical-reference/build-and-push-to-ecr-step-settings.md [hunk 2]
```diff
@@ -24,6 +24,14 @@ See [Entity Identifier Reference](../../platform/20_References/entity-identifier
...
 The Harness AWS Connector to use to connect to ECR. The AWS IAM roles and policies associated with the account used in the Harness AWS Connector must be able to push to ECR. See [AWS Connector Settings Reference](../../platform/7_Connectors/ref-cloud-providers/aws-connector-settings-reference.md).
 
+
+:::note
+
+This step supports AWS connectors with any authentication method (AWS access key, Delegate IAM role assumption, IRSA, and cross account access).
+
+:::
+
+
```
Comment (aimurphy) [inline docs/continuous-integration/ci-technical-reference/build-and-push-to-ecr-step-settings.md:34]: CI-6410

File: docs/continuous-integration/ci-technical-reference/build-and-push-to-gcr-step-settings.md [hunk 3]
```diff
@@ -26,6 +26,12 @@ See [Entity Identifier Reference](../../platform/20_References/entity-identifier
 
 The Harness GCP Connector to use to connect to GCR. See [Google Cloud Platform (GCP) Connector Settings Reference](../../platform/7_Connectors/ref-cloud-providers/gcs-connector-settings-reference.md).
 
+:::note
+
+This step has been tested with GCP connectors that use access key authentication. GCP connectors that inherit delegate credentials may encounter problems during pipeline execution.
+
+:::
+
```
Comment (aimurphy) [inline docs/continuous-integration/ci-technical-reference/build-and-push-to-gcr-step-settings.md:34]: CI-6410

File: docs/continuous-integration/ci-technical-reference/restore-cache-from-gcs-settings.md [hunk 4]
```diff
@@ -22,6 +22,12 @@ See [Entity Identifier Reference](../../platform/20_References/entity-identifier
 
 The Harness Connector for the GCP account where you saved the cache.
 
+:::note
+
+This step has been tested with GCP connectors that use access key authentication. GCP connectors that inherit delegate credentials may encounter problems during pipeline execution.
+
+:::
+
```
Comment (aimurphy) [inline docs/continuous-integration/ci-technical-reference/restore-cache-from-gcs-settings.md:30]: CI-6410

File: docs/continuous-integration/ci-technical-reference/restore-cache-from-s-3-step-settings.md [hunk 5]
```diff
@@ -24,6 +24,16 @@ The Harness Connector to use when restoring the cache from AWS S3. Typically, th
...
+:::note
+
+This step supports AWS connectors using **AWS Access Key** and **Assume IAM role on Delegate** authentication methods *without* cross account access (ARN/STS).
+
+AWS connectors using IRSA authentication may encounter problems during pipeline execution.
+
+This step does not support AWS connectors that have enabled cross account access (ARN/STS), regardless of authentication method.
+
+:::
+
```
Comment (aimurphy) [inline docs/continuous-integration/ci-technical-reference/restore-cache-from-s-3-step-settings.md:36]: CI-6410

File: docs/continuous-integration/ci-technical-reference/save-cache-to-gcs-step-settings.md [hunk 6]
```diff
@@ -22,6 +22,12 @@ See [Entity Identifier Reference](../../platform/20_References/entity-identifier
 
 The Harness Connector for the GCP account where you want to save the cache.
 
+:::note
+
+This step has been tested with GCP connectors that use access key authentication. GCP connectors that inherit delegate credentials may encounter problems during pipeline execution.
+
+:::
+
```
Comment (aimurphy) [inline docs/continuous-integration/ci-technical-reference/save-cache-to-gcs-step-settings.md:30]: CI-6410

File: docs/continuous-integration/ci-technical-reference/save-cache-to-s-3-step-settings.md [hunk 7]
```diff
@@ -24,6 +24,16 @@ See [Entity Identifier Reference](../../platform/20_References/entity-identifier
...
+:::note
+
+This step supports AWS connectors using **AWS Access Key** and **Assume IAM role on Delegate** authentication methods *without* cross account access (ARN/STS).
+
+AWS connectors using IRSA authentication may encounter problems during pipeline execution.
+
+This step does not support AWS connectors that have enabled cross account access (ARN/STS), regardless of authentication method.
+
+:::
+
```
Comment (aimurphy) [inline docs/continuous-integration/ci-technical-reference/save-cache-to-s-3-step-settings.md:36]: CI-6410

File: docs/continuous-integration/ci-technical-reference/upload-artifacts-to-gcs-step-settings.md [hunk 8]
```diff
@@ -22,6 +22,12 @@ See [Entity Identifier Reference](../../platform/20_References/entity-identifier
 
 The Harness Connector for the GCP account where you want to upload the artifact. See [Google Cloud Platform (GCP) Connector Settings Reference](../../platform/7_Connectors/ref-cloud-providers/gcs-connector-settings-reference.md).
 
+:::note
+
+This step has been tested with GCP connectors that use access key authentication. GCP connectors that inherit delegate credentials may encounter problems during pipeline execution.
+
+:::
+
```
Comment (aimurphy) [inline docs/continuous-integration/ci-technical-reference/upload-artifacts-to-gcs-step-settings.md:30]: CI-6410

File: docs/continuous-integration/ci-technical-reference/upload-artifacts-to-jfrog-artifactory-step-settings.md [hunk 9]
```diff
@@ -24,6 +24,12 @@ Select the Harness Artifactory Connector to use for this upload.
 
 See [Artifactory Connector Settings Reference](../../platform/7_Connectors/ref-cloud-providers/artifactory-connector-settings-reference.md).
 
+:::note
+
+This step supports Artifactory connectors that use either anonymous or username and password authentication.
+
+:::
+
```
Comment (aimurphy) [inline docs/continuous-integration/ci-technical-reference/upload-artifacts-to-jfrog-artifactory-step-settings.md:32]: CI-6410

File: docs/continuous-integration/ci-technical-reference/upload-artifacts-to-s-3-step-settings.md [hunk 10]
```diff
@@ -1,75 +1,97 @@
...
 
-Path to the artifact files you want to upload. You can use standard [glob expressions](https://en.wikipedia.org/wiki/Glob_(programming)) to upload multiple files. For example, `src/js/**/*.js` will upload all Javascript files in `src/js/subfolder-1/`, `src/js/subfolder-2`, and so on.
+1. In the Pipeline Studio, select the stage with the **Upload Artifacts to S3** step, and then select the **Overview** tab.
+2. In the **Advanced** section, add a stage variable.
+3. Input `PLUGIN_USER_ROLE_ARN` as the **Variable Name**, set the **Type** to **String**, and then select **Save**.
+4. In the **Value** field, input the full ARN value that corresponds with the AWS connector's ARN.
+```
 
-### Optional Configuration
+</details>
```
Comment (aimurphy) [inline docs/continuous-integration/ci-technical-reference/upload-artifacts-to-s-3-step-settings.md:52]: DOC-2351 @ken-harness

**Review by rajatharanganath — APPROVED** [review]

**Review by rajatharanganath — APPROVED** [review]

File: docs/platform/7_Connectors/ref-cloud-providers/gcs-connector-settings-reference.md [hunk 11]
```diff
@@ -1,101 +1,116 @@
 ---
-title: Google Cloud Platform (GCP) Connector Settings Reference
-description: This topic provides settings and permissions for the GCP Connector.
+title: Google Cloud Platform (GCP) connector settings reference
+description: This topic provides settings and permissions for the GCP connector.
```
Comment (michaelcretzman) [inline docs/platform/7_Connectors/ref-cloud-providers/gcs-connector-settings-reference.md:3]:
```suggestion
description: Settings and permissions for the GCP connector.
```

**Review by michaelcretzman — APPROVED** [review]
approved with the assumption that you'll consider my suggestions.

**Comment by aimurphy** [comment]
Force push was from a rebase to resolve merge conflicts.

**Comment by aimurphy** [comment]
Michael has approved for peer review. Will merge after preview build succeeds.

**Pull request merged**